docs/source/weights_loading.rst

```diff
@@ -59,24 +59,20 @@ Or disable it by passing
    trainer = Trainer(checkpoint_callback=False)
 
 
-The Lightning checkpoint also saves the hparams (hyperparams) passed into the LightningModule init.
+The Lightning checkpoint also saves the arguments passed into the LightningModule init
+under the `module_arguments` key in the checkpoint.
 
-.. note:: hparams is a `Namespace <https://docs.python.org/2/library/argparse.html#argparse.Namespace>`_.
-
-.. testcode::
-
-   from argparse import Namespace
+.. code-block:: python
 
-   # usually these come from command line args
-   args = Namespace(learning_rate=0.001)
+    class MyLightningModule(LightningModule):
 
-   # define you module to have hparams as the first arg
-   # this means your checkpoint will have everything that went into making
-   # this model (in this case, learning rate)
-   class MyLightningModule(LightningModule):
+       def __init__(self, learning_rate, *args, **kwargs):
+            super().__init__()
 
-       def __init__(self, hparams, *args, **kwargs):
-           self.hparams = hparams
+    # all init args were saved to the checkpoint
+    checkpoint = torch.load(CKPT_PATH)
+    print(checkpoint[CHECKPOINT_KEY_MODULE_ARGS])
```

Comment:
@awaelchli @Borda users are reading this. Let's use the actual names instead of the enums. Otherwise this wasn't informative as they now have to go figure out the key